plugins/outputs/pulsar/pulsar.go

```diff
@@ -0,0 +1,378 @@
+ ## Each data format has its own unique set of configuration options, read
+ ## more about them here:
+ ## https://github.com/influxdata/telegraf/blob/master/docs/DATA_FORMATS_OUTPUT.md
+ # data_format = "influx"
+
+`
+
+func ValidateTopicSuffixMethod(method string) error {
+	for _, validMethod := range ValidTopicSuffixMethods {
+		if method == validMethod {
+			return nil
+		}
+	}
+	return fmt.Errorf("Unknown topic suffix method provided: %s", method)
+}
+
+func (p *Pulsar) GetTopicName(metric telegraf.Metric) (telegraf.Metric, string) {
+	topic := p.Topic
+	if p.TopicTag != "" {
+		if t, ok := metric.GetTag(p.TopicTag); ok {
+			topic = p.tenantNameSpace + "/" + t
+			// If excluding the topic tag, a copy is required to avoid modifying
+			// the metric buffer.
+			if p.ExcludeTopicTag {
+				metric = metric.Copy()
+				metric.Accept()
+				metric.RemoveTag(p.TopicTag)
+			}
+		}
+	}
+	return metric, topic
+}
+
+func (p *Pulsar) GetProducer(topic string, metricBatchSize uint) (pulsar.Producer, error) {
+	if producer, ok := p.producerCache[topic]; !ok {
+		producerOptions := pulsar.ProducerOptions{
+			Topic:               topic,
+			BatchingMaxMessages: metricBatchSize,
+		}
+		switch p.CompressionType {
+		case 0:
+			producerOptions.CompressionType = pulsar.NoCompression
+		case 1:
+			producerOptions.CompressionType = pulsar.LZ4
+		case 2:
+			producerOptions.CompressionType = pulsar.ZLib
+		case 4:
+			producerOptions.CompressionType = pulsar.ZSTD
+		default:
+			producerOptions.CompressionType = pulsar.NoCompression
+		}
+
+		switch p.HashingScheme {
+		case "Murmur3_32Hash":
+			producerOptions.HashingScheme = pulsar.Murmur3_32Hash
+		case "JavaStringHash":
+			producerOptions.HashingScheme = pulsar.JavaStringHash
+		default:
+			producerOptions.HashingScheme = pulsar.JavaStringHash
+		}
+		if p.MaxPendingMessages > 0 {
+			producerOptions.MaxPendingMessages = p.MaxPendingMessages
+		}
+		duration, err := time.ParseDuration(p.BatchingMaxPublishDelay)
+		if err != nil {
+			producerOptions.BatchingMaxPublishDelay = duration
+		}
+		producer, err := p.client.CreateProducer(producerOptions)
+		if err != nil {
+			return nil, err
+		} else {
+			p.Log.Infof("Created producer for topic %s", topic)
+			p.producerCache[topic] = producer
+			return producer, nil
+		}
+
+	} else {
+		p.Log.Debugf("Returning cached producer for topic %s", topic)
+		return producer, nil
+	}
+
+}
+func (p *Pulsar) SetSerializer(serializer serializers.Serializer) {
+	p.serializer = serializer
+}
+
+func (p *Pulsar) Connect() error {
+	p.producerCache = make(map[string]pulsar.Producer)
+	splits := strings.Split(p.Topic, "/")
+	p.tenantNameSpace = strings.Join(splits[0:4], "/")
+	clientOptions := pulsar.ClientOptions{
+		URL:                        p.URL,
+		TLSAllowInsecureConnection: p.TLSAllowInsecureConnection,
+		TLSTrustCertsFilePath:      p.TLSTrustCertsFilePath,
+		TLSValidateHostname:        p.TLSValidateHostname,
+	}
+	if p.ConnectionTimeout != "" {
```

Comment:
`ConnectionTimeout` will be `""` in case we comment the option and use the default. Therefore, the user expects that, in case the option is commented you use the default. The best way is to initialize `ConnectionTimeout` when creating the struct instance in `init()`.